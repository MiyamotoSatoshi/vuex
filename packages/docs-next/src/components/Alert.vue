<template>
  <v-alert
    border="left"
    text
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot />
  </v-alert>
</template>

<script>
  export default { name: 'Alert' }
</script>
