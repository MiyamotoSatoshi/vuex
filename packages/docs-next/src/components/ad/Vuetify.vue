<template>
  <app-ad
    :color="bgColor"
    v-bind="$props"
  >
    <v-list-item
      v-if="current"
      :color="bgColor"
      class="pl-2 rounded"
      v-bind="adAttrs"
    >
      <v-list-item-avatar
        v-if="metadata.src && !compact"
        size="56"
        tile
      >
        <v-img
          :alt="`Link to ${current.title}`"
          :src="metadata.src"
          class="rounded-tl rounded-bl"
        />
      </v-list-item-avatar>

      <v-icon
        v-else-if="icon"
        :class="[color]"
        class="mr-3"
        large
        v-text="icon"
      />

      <v-list-item-content>
        <v-list-item-title
          v-if="!isSponsored"
          class="font-weight-medium mb-1 subtitle-1"
          v-text="current.title"
        />

        <v-list-item-subtitle
          v-if="metadata.description"
          :class="[color, isSponsored && 'body-2 font-weight-medium']"
          v-text="metadata.description"
        />
      </v-list-item-content>

      <v-list-item-action
        v-if="!compact"
        class="mb-n8 mr-n2"
      >
        <span class="overline text--secondary">ads via Vuetify</span>
      </v-list-item-action>
    </v-list-item>
  </app-ad>
</template>

<script>
  // Utilities
  import { get } from 'vuex-pathify'

  export default {
    name: 'VuetifyAd',

    props: {
      bgColor: String,
      color: String,
      comfortable: Boolean,
      compact: Boolean,
      discover: Boolean,
      outlined: Boolean,
      slug: String,
    },

    computed: {
      all: get('ads/all'),
      ads () {
        if (!this.discover) return this.all

        return this.all.filter(ad => ad.metadata.type === 'Discovery')
      },
      adAttrs () {
        if (!this.current) return undefined

        const [url, query] = this.metadata.url.split('?')

        if (!url.startsWith('http')) {
          return { to: `/${this.lang}${url}/` }
        }

        return {
          href: `${url}?ref=vuetifyjs.com${query ? `&${query}` : ''}`,
          target: '_blank',
        }
      },
      icon () {
        switch (this.metadata.type) {
          case 'Video': return '$mdiPlayCircle'
          default: return '$mdiVuetify'
        }
      },
      isSponsored () {
        return this.metadata.sponsored
      },
      current () {
        const index = !this.slug
          ? this.getRandomIndex()
          : this.getSlugIndex()

        return this.ads[index]
      },
      metadata () {
        return this.current ? this.current.metadata : {}
      },
    },

    methods: {
      getSlugIndex () {
        return this.ads.findIndex(ad => ad.slug === this.slug)
      },
      getRandomIndex () {
        return Math.floor(Math.random() * this.ads.length)
      },
    },
  }
</script>
