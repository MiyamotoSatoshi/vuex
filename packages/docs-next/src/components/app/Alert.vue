<template>
  <v-alert
    v-bind="$attrs"
    text
    v-on="$listeners"
  >
    <slot />
  </v-alert>
</template>

<script>
  export default {
    name: 'AppAlert',
  }
</script>
