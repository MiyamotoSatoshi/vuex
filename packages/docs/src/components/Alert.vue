<template>
  <v-alert
    border="left"
    class="v-alert--doc"
    text
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot />
  </v-alert>
</template>

<script>
  export default {
    name: 'Alert',
  }
</script>

<style lang="sass">
  .v-alert--doc
    .v-alert__content p
      margin-bottom: 0 !important
</style>
