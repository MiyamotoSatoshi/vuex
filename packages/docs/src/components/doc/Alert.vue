<template>
  <v-alert
    :class="$vuetify.theme.dark ? undefined : 'grey lighten-3'"
    :type="value"
    border="left"
    class="app-alert mb-4"
    colored-border
    value
  >
    <base-markdown><slot /></base-markdown>
  </v-alert>
</template>

<script>
  export default {
    name: 'AppAlert',

    props: {
      value: {
        type: String,
        default: '',
      },
    },
  }
</script>

<style lang="sass">
.app-alert
  p
    margin: 0 !important
</style>
