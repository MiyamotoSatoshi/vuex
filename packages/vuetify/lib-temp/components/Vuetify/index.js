import OurVue from 'vue';
import application from './mixins/application';
import breakpoint from './mixins/breakpoint';
import theme from './mixins/theme';
import icons from './mixins/icons';
import options from './mixins/options';
import genLang from './mixins/lang';
import goTo from './util/goTo';
// Utils
import { consoleWarn, consoleError } from '../../util/console';
const Vuetify = {
    install(Vue, opts = {}) {
        if (this.installed)
            return;
        this.installed = true;
        if (OurVue !== Vue) {
            consoleError('Multiple instances of Vue detected\nSee https://github.com/vuetifyjs/vuetify/issues/4068\n\nIf you\'re seeing "$attrs is readonly", it\'s caused by this');
        }
        checkVueVersion(Vue);
        const lang = genLang(opts.lang);
        Vue.prototype.$vuetify = new Vue({
            mixins: [
                breakpoint
            ],
            data: {
                application,
                dark: false,
                icons: icons(opts.iconfont, opts.icons),
                lang,
                options: options(opts.options),
                rtl: opts.rtl,
                theme: theme(opts.theme)
            },
            methods: {
                goTo,
                t: lang.t.bind(lang)
            }
        });
        if (opts.directives) {
            for (const name in opts.directives) {
                Vue.directive(name, opts.directives[name]);
            }
        }
        (function registerComponents(components) {
            if (components) {
                for (const key in components) {
                    const component = components[key];
                    if (component && !registerComponents(component.$_vuetify_subcomponents)) {
                        Vue.component(key, component);
                    }
                }
                return true;
            }
            return false;
        })(opts.components);
    },
    version: __VUETIFY_VERSION__
};
export function checkVueVersion(Vue, requiredVue) {
    const vueDep = requiredVue || __REQUIRED_VUE__;
    const required = vueDep.split('.', 3).map(v => v.replace(/\D/g, '')).map(Number);
    const actual = Vue.version.split('.', 3).map(n => parseInt(n, 10));
    // Simple semver caret range comparison
    const passes = actual[0] === required[0] && // major matches
        (actual[1] > required[1] || // minor is greater
            (actual[1] === required[1] && actual[2] >= required[2]) // or minor is eq and patch is >=
        );
    if (!passes) {
        consoleWarn(`Vuetify requires Vue version ${vueDep}`);
    }
}
export default Vuetify;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WdWV0aWZ5L2luZGV4LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sTUFBTSxNQUFNLEtBQUssQ0FBQTtBQUV4QixPQUFPLFdBQVcsTUFBTSxzQkFBc0IsQ0FBQTtBQUM5QyxPQUFPLFVBQVUsTUFBTSxxQkFBcUIsQ0FBQTtBQUM1QyxPQUFPLEtBQUssTUFBTSxnQkFBZ0IsQ0FBQTtBQUNsQyxPQUFPLEtBQUssTUFBTSxnQkFBZ0IsQ0FBQTtBQUNsQyxPQUFPLE9BQU8sTUFBTSxrQkFBa0IsQ0FBQTtBQUN0QyxPQUFPLE9BQU8sTUFBTSxlQUFlLENBQUE7QUFDbkMsT0FBTyxJQUFJLE1BQU0sYUFBYSxDQUFBO0FBRTlCLFFBQVE7QUFDUixPQUFPLEVBQUUsV0FBVyxFQUFFLFlBQVksRUFBRSxNQUFNLG9CQUFvQixDQUFBO0FBTTlELE1BQU0sT0FBTyxHQUFrQjtJQUM3QixPQUFPLENBQUUsR0FBRyxFQUFFLElBQUksR0FBRyxFQUFFO1FBQ3JCLElBQUssSUFBWSxDQUFDLFNBQVM7WUFBRSxPQUM1QjtRQUFDLElBQVksQ0FBQyxTQUFTLEdBQUcsSUFBSSxDQUFBO1FBRS9CLElBQUksTUFBTSxLQUFLLEdBQUcsRUFBRTtZQUNsQixZQUFZLENBQUMsMEpBQTBKLENBQUMsQ0FBQTtTQUN6SztRQUVELGVBQWUsQ0FBQyxHQUFHLENBQUMsQ0FBQTtRQUVwQixNQUFNLElBQUksR0FBRyxPQUFPLENBQUMsSUFBSSxDQUFDLElBQUksQ0FBQyxDQUFBO1FBRS9CLEdBQUcsQ0FBQyxTQUFTLENBQUMsUUFBUSxHQUFHLElBQUksR0FBRyxDQUFDO1lBQy9CLE1BQU0sRUFBRTtnQkFDTixVQUFVO2FBQ1g7WUFDRCxJQUFJLEVBQUU7Z0JBQ0osV0FBVztnQkFDWCxJQUFJLEVBQUUsS0FBSztnQkFDWCxLQUFLLEVBQUUsS0FBSyxDQUFDLElBQUksQ0FBQyxRQUFRLEVBQUUsSUFBSSxDQUFDLEtBQUssQ0FBQztnQkFDdkMsSUFBSTtnQkFDSixPQUFPLEVBQUUsT0FBTyxDQUFDLElBQUksQ0FBQyxPQUFPLENBQUM7Z0JBQzlCLEdBQUcsRUFBRSxJQUFJLENBQUMsR0FBRztnQkFDYixLQUFLLEVBQUUsS0FBSyxDQUFDLElBQUksQ0FBQyxLQUFLLENBQUM7YUFDekI7WUFDRCxPQUFPLEVBQUU7Z0JBQ1AsSUFBSTtnQkFDSixDQUFDLEVBQUUsSUFBSSxDQUFDLENBQUMsQ0FBQyxJQUFJLENBQUMsSUFBSSxDQUFDO2FBQ3JCO1NBQ0YsQ0FBQyxDQUFBO1FBRUYsSUFBSSxJQUFJLENBQUMsVUFBVSxFQUFFO1lBQ25CLEtBQUssTUFBTSxJQUFJLElBQUksSUFBSSxDQUFDLFVBQVUsRUFBRTtnQkFDbEMsR0FBRyxDQUFDLFNBQVMsQ0FBQyxJQUFJLEVBQUUsSUFBSSxDQUFDLFVBQVUsQ0FBQyxJQUFJLENBQUMsQ0FBQyxDQUFBO2FBQzNDO1NBQ0Y7UUFFRCxDQUFDLFNBQVMsa0JBQWtCLENBQUUsVUFBMkM7WUFDdkUsSUFBSSxVQUFVLEVBQUU7Z0JBQ2QsS0FBSyxNQUFNLEdBQUcsSUFBSSxVQUFVLEVBQUU7b0JBQzVCLE1BQU0sU0FBUyxHQUFHLFVBQVUsQ0FBQyxHQUFHLENBQUMsQ0FBQTtvQkFDakMsSUFBSSxTQUFTLElBQUksQ0FBQyxrQkFBa0IsQ0FBQyxTQUFTLENBQUMsdUJBQXVCLENBQUMsRUFBRTt3QkFDdkUsR0FBRyxDQUFDLFNBQVMsQ0FBQyxHQUFHLEVBQUUsU0FBdUIsQ0FBQyxDQUFBO3FCQUM1QztpQkFDRjtnQkFDRCxPQUFPLElBQUksQ0FBQTthQUNaO1lBQ0QsT0FBTyxLQUFLLENBQUE7UUFDZCxDQUFDLENBQUMsQ0FBQyxJQUFJLENBQUMsVUFBVSxDQUFDLENBQUE7SUFDckIsQ0FBQztJQUNELE9BQU8sRUFBRSxtQkFBbUI7Q0FDN0IsQ0FBQTtBQUVELE1BQU0sVUFBVSxlQUFlLENBQUUsR0FBbUIsRUFBRSxXQUFvQjtJQUN4RSxNQUFNLE1BQU0sR0FBRyxXQUFXLElBQUksZ0JBQWdCLENBQUE7SUFFOUMsTUFBTSxRQUFRLEdBQUcsTUFBTSxDQUFDLEtBQUssQ0FBQyxHQUFHLEVBQUUsQ0FBQyxDQUFDLENBQUMsR0FBRyxDQUFDLENBQUMsQ0FBQyxFQUFFLENBQUMsQ0FBQyxDQUFDLE9BQU8sQ0FBQyxLQUFLLEVBQUUsRUFBRSxDQUFDLENBQUMsQ0FBQyxHQUFHLENBQUMsTUFBTSxDQUFDLENBQUE7SUFDaEYsTUFBTSxNQUFNLEdBQUcsR0FBRyxDQUFDLE9BQU8sQ0FBQyxLQUFLLENBQUMsR0FBRyxFQUFFLENBQUMsQ0FBQyxDQUFDLEdBQUcsQ0FBQyxDQUFDLENBQUMsRUFBRSxDQUFDLFFBQVEsQ0FBQyxDQUFDLEVBQUUsRUFBRSxDQUFDLENBQUMsQ0FBQTtJQUVsRSx1Q0FBdUM7SUFDdkMsTUFBTSxNQUFNLEdBQ1YsTUFBTSxDQUFDLENBQUMsQ0FBQyxLQUFLLFFBQVEsQ0FBQyxDQUFDLENBQUMsSUFBSSxnQkFBZ0I7UUFDN0MsQ0FBQyxNQUFNLENBQUMsQ0FBQyxDQUFDLEdBQUcsUUFBUSxDQUFDLENBQUMsQ0FBQyxJQUFJLG1CQUFtQjtZQUM3QyxDQUFDLE1BQU0sQ0FBQyxDQUFDLENBQUMsS0FBSyxRQUFRLENBQUMsQ0FBQyxDQUFDLElBQUksTUFBTSxDQUFDLENBQUMsQ0FBQyxJQUFJLFFBQVEsQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDLGlDQUFpQztTQUMxRixDQUFBO0lBRUgsSUFBSSxDQUFDLE1BQU0sRUFBRTtRQUNYLFdBQVcsQ0FBQyxnQ0FBZ0MsTUFBTSxFQUFFLENBQUMsQ0FBQTtLQUN0RDtBQUNILENBQUM7QUFFRCxlQUFlLE9BQU8sQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBPdXJWdWUgZnJvbSAndnVlJ1xuXG5pbXBvcnQgYXBwbGljYXRpb24gZnJvbSAnLi9taXhpbnMvYXBwbGljYXRpb24nXG5pbXBvcnQgYnJlYWtwb2ludCBmcm9tICcuL21peGlucy9icmVha3BvaW50J1xuaW1wb3J0IHRoZW1lIGZyb20gJy4vbWl4aW5zL3RoZW1lJ1xuaW1wb3J0IGljb25zIGZyb20gJy4vbWl4aW5zL2ljb25zJ1xuaW1wb3J0IG9wdGlvbnMgZnJvbSAnLi9taXhpbnMvb3B0aW9ucydcbmltcG9ydCBnZW5MYW5nIGZyb20gJy4vbWl4aW5zL2xhbmcnXG5pbXBvcnQgZ29UbyBmcm9tICcuL3V0aWwvZ29UbydcblxuLy8gVXRpbHNcbmltcG9ydCB7IGNvbnNvbGVXYXJuLCBjb25zb2xlRXJyb3IgfSBmcm9tICcuLi8uLi91dGlsL2NvbnNvbGUnXG5cbi8vIFR5cGVzXG5pbXBvcnQgeyBWdWVDb25zdHJ1Y3RvciB9IGZyb20gJ3Z1ZS90eXBlcydcbmltcG9ydCB7IFZ1ZXRpZnkgYXMgVnVldGlmeVBsdWdpbiwgVnVldGlmeVVzZU9wdGlvbnMgfSBmcm9tICd0eXBlcydcblxuY29uc3QgVnVldGlmeTogVnVldGlmeVBsdWdpbiA9IHtcbiAgaW5zdGFsbCAoVnVlLCBvcHRzID0ge30pIHtcbiAgICBpZiAoKHRoaXMgYXMgYW55KS5pbnN0YWxsZWQpIHJldHVyblxuICAgIDsodGhpcyBhcyBhbnkpLmluc3RhbGxlZCA9IHRydWVcblxuICAgIGlmIChPdXJWdWUgIT09IFZ1ZSkge1xuICAgICAgY29uc29sZUVycm9yKCdNdWx0aXBsZSBpbnN0YW5jZXMgb2YgVnVlIGRldGVjdGVkXFxuU2VlIGh0dHBzOi8vZ2l0aHViLmNvbS92dWV0aWZ5anMvdnVldGlmeS9pc3N1ZXMvNDA2OFxcblxcbklmIHlvdVxcJ3JlIHNlZWluZyBcIiRhdHRycyBpcyByZWFkb25seVwiLCBpdFxcJ3MgY2F1c2VkIGJ5IHRoaXMnKVxuICAgIH1cblxuICAgIGNoZWNrVnVlVmVyc2lvbihWdWUpXG5cbiAgICBjb25zdCBsYW5nID0gZ2VuTGFuZyhvcHRzLmxhbmcpXG5cbiAgICBWdWUucHJvdG90eXBlLiR2dWV0aWZ5ID0gbmV3IFZ1ZSh7XG4gICAgICBtaXhpbnM6IFtcbiAgICAgICAgYnJlYWtwb2ludFxuICAgICAgXSxcbiAgICAgIGRhdGE6IHtcbiAgICAgICAgYXBwbGljYXRpb24sXG4gICAgICAgIGRhcms6IGZhbHNlLFxuICAgICAgICBpY29uczogaWNvbnMob3B0cy5pY29uZm9udCwgb3B0cy5pY29ucyksXG4gICAgICAgIGxhbmcsXG4gICAgICAgIG9wdGlvbnM6IG9wdGlvbnMob3B0cy5vcHRpb25zKSxcbiAgICAgICAgcnRsOiBvcHRzLnJ0bCxcbiAgICAgICAgdGhlbWU6IHRoZW1lKG9wdHMudGhlbWUpXG4gICAgICB9LFxuICAgICAgbWV0aG9kczoge1xuICAgICAgICBnb1RvLFxuICAgICAgICB0OiBsYW5nLnQuYmluZChsYW5nKVxuICAgICAgfVxuICAgIH0pXG5cbiAgICBpZiAob3B0cy5kaXJlY3RpdmVzKSB7XG4gICAgICBmb3IgKGNvbnN0IG5hbWUgaW4gb3B0cy5kaXJlY3RpdmVzKSB7XG4gICAgICAgIFZ1ZS5kaXJlY3RpdmUobmFtZSwgb3B0cy5kaXJlY3RpdmVzW25hbWVdKVxuICAgICAgfVxuICAgIH1cblxuICAgIChmdW5jdGlvbiByZWdpc3RlckNvbXBvbmVudHMgKGNvbXBvbmVudHM6IFZ1ZXRpZnlVc2VPcHRpb25zWydjb21wb25lbnRzJ10pIHtcbiAgICAgIGlmIChjb21wb25lbnRzKSB7XG4gICAgICAgIGZvciAoY29uc3Qga2V5IGluIGNvbXBvbmVudHMpIHtcbiAgICAgICAgICBjb25zdCBjb21wb25lbnQgPSBjb21wb25lbnRzW2tleV1cbiAgICAgICAgICBpZiAoY29tcG9uZW50ICYmICFyZWdpc3RlckNvbXBvbmVudHMoY29tcG9uZW50LiRfdnVldGlmeV9zdWJjb21wb25lbnRzKSkge1xuICAgICAgICAgICAgVnVlLmNvbXBvbmVudChrZXksIGNvbXBvbmVudCBhcyB0eXBlb2YgVnVlKVxuICAgICAgICAgIH1cbiAgICAgICAgfVxuICAgICAgICByZXR1cm4gdHJ1ZVxuICAgICAgfVxuICAgICAgcmV0dXJuIGZhbHNlXG4gICAgfSkob3B0cy5jb21wb25lbnRzKVxuICB9LFxuICB2ZXJzaW9uOiBfX1ZVRVRJRllfVkVSU0lPTl9fXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBjaGVja1Z1ZVZlcnNpb24gKFZ1ZTogVnVlQ29uc3RydWN0b3IsIHJlcXVpcmVkVnVlPzogc3RyaW5nKSB7XG4gIGNvbnN0IHZ1ZURlcCA9IHJlcXVpcmVkVnVlIHx8IF9fUkVRVUlSRURfVlVFX19cblxuICBjb25zdCByZXF1aXJlZCA9IHZ1ZURlcC5zcGxpdCgnLicsIDMpLm1hcCh2ID0+IHYucmVwbGFjZSgvXFxEL2csICcnKSkubWFwKE51bWJlcilcbiAgY29uc3QgYWN0dWFsID0gVnVlLnZlcnNpb24uc3BsaXQoJy4nLCAzKS5tYXAobiA9PiBwYXJzZUludChuLCAxMCkpXG5cbiAgLy8gU2ltcGxlIHNlbXZlciBjYXJldCByYW5nZSBjb21wYXJpc29uXG4gIGNvbnN0IHBhc3NlcyA9XG4gICAgYWN0dWFsWzBdID09PSByZXF1aXJlZFswXSAmJiAvLyBtYWpvciBtYXRjaGVzXG4gICAgKGFjdHVhbFsxXSA+IHJlcXVpcmVkWzFdIHx8IC8vIG1pbm9yIGlzIGdyZWF0ZXJcbiAgICAgIChhY3R1YWxbMV0gPT09IHJlcXVpcmVkWzFdICYmIGFjdHVhbFsyXSA+PSByZXF1aXJlZFsyXSkgLy8gb3IgbWlub3IgaXMgZXEgYW5kIHBhdGNoIGlzID49XG4gICAgKVxuXG4gIGlmICghcGFzc2VzKSB7XG4gICAgY29uc29sZVdhcm4oYFZ1ZXRpZnkgcmVxdWlyZXMgVnVlIHZlcnNpb24gJHt2dWVEZXB9YClcbiAgfVxufVxuXG5leHBvcnQgZGVmYXVsdCBWdWV0aWZ5XG4iXX0=