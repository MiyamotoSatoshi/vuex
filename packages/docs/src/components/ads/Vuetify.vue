<template>
  <ads-base
    class="v-vuetify"
    v-bind="$attrs"
    comfortable
    border
  >
    <v-list-item
      v-if="ad"
      class="rounded px-2"
      style="min-height: inherit; width: 100%"
      v-bind="attrs"
      :prepend-avatar="src"
      :title="ad.title"
    >
      <template #subtitle>
        <app-markdown
          v-if="description"
          class="text-caption text--secondary"
          :content="description"
        />
      </template>
    </v-list-item>
  </ads-base>
</template>

<script lang="ts">
  import { defineComponent } from 'vue'
  import { useI18n } from 'vue-i18n'
  import { createAdProps, useAd } from '../../composables/ad'

  import AdsBase from './Base.vue'

  // Mixins
  export default defineComponent({
    name: 'AdsVuetify',

    components: { AdsBase },

    props: {
      color: String,

      ...createAdProps(),
    },

    setup (props) {
      const { t } = useI18n()
      const { ad, attrs, src, description } = useAd(props)

      return { t, ad, attrs, src, description }
    },
  })
</script>

<style lang="sass">
  .v-vuetify
    .powered-by
      color: rgba(0, 0, 0, .6)
      font-size: 0.625rem
      font-weight: 400
      letter-spacing: 0.09375rem
      text-transform: uppercase

    &.theme--dark .powered-by
      color: inherit
</style>
