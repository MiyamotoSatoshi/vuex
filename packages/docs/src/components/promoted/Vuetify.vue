<template>
  <promoted-base
    border
    class="v-vuetify mb-4"
    density="comfortable"
  >
    <v-list-item
      v-if="ad"
      :prepend-avatar="src"
      :title="ad.title"
      append-icon="mdi-open-in-new"
      style="min-height: inherit; width: 100%"
      v-bind="attrs"
    >
      <template #subtitle>
        <app-markdown
          v-if="description"
          :content="description"
          class="text-caption"
        />
      </template>
    </v-list-item>
  </promoted-base>
</template>

<script setup>
  // Components
  import PromotedBase from './Base.vue'

  // Composables
  import { createAdProps, useAd } from '@/composables/ad'

  const props = defineProps({
    color: String,

    ...createAdProps(),
  })

  const { ad, attrs, src, description } = useAd(props)
</script>

<style lang="sass">
  .v-vuetify
    .powered-by
      color: rgba(0, 0, 0, .6)
      font-size: 0.625rem
      font-weight: 400
      letter-spacing: 0.09375rem
      text-transform: uppercase

    &.theme--dark .powered-by
      color: inherit
</style>
