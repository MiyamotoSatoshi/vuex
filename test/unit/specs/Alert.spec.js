import { mount } from 'avoriaz'
import { createRenderer } from 'vue-server-renderer'
import Vue from 'vue/dist/vue.common'
import Vuetify from 'src/index'
import Alert from 'src/components/alerts/Alert'

Vue.use(Vuetify)

describe('Alert.vue', () => {

  it('should have an alert class', () => {
    const wrapper = mount(Alert)

    expect(wrapper.hasClass('alert')).toBe(true)
  })

  it('should render correct contents', () => {
    const wrapper = mount(Alert)
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

})
