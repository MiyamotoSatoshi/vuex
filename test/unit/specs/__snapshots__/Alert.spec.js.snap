// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert.vue should render correct contents 1`] = `"<div data-server-rendered=\\"true\\" class=\\"alert\\" style=\\"display:none;\\"><div></div></div>"`;
